esp-hal/src/uart.rs

```diff
@@ -1424,9 +1424,58 @@ mod asynch {
         }
     }
 
-    const INIT: AtomicWaker = AtomicWaker::new();
-    static TX_WAKERS: [AtomicWaker; NUM_UART] = [INIT; NUM_UART];
-    static RX_WAKERS: [AtomicWaker; NUM_UART] = [INIT; NUM_UART];
+    const WAKER_INIT: AtomicWaker = AtomicWaker::new();
+    static TX_WAKERS: [AtomicWaker; NUM_UART] = [WAKER_INIT; NUM_UART];
+    static RX_WAKERS: [AtomicWaker; NUM_UART] = [WAKER_INIT; NUM_UART];
+
+    const BUFFER_INIT: Option<&mut dyn RxBuffer> = None;
+    static mut RX_BUFFERS: [Option<&mut dyn RxBuffer>; NUM_UART] = [BUFFER_INIT; NUM_UART];
+
+    pub (crate) trait RxBuffer {
```

Comment:
Is there any reason to make this a trait over a struct containing a `heapless::Deque`?